internal/client/client_cancel_test.go

```
@@ -0,0 +1,161 @@
+}
+
+// A ^C during an upload must abort promptly even though bytes were still
+// flowing, which the stall guard alone would never do.
+func TestCancelDuringUploadAborts(t *testing.T) {
+	t.Parallel()
+	uploading := make(chan struct{})
+	done := make(chan struct{})
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Read a little so the client is genuinely mid-upload, then stop draining.
+		buf := make([]byte, 1024)
+		r.Body.Read(buf)
```

golangci-lint (2.12.2)

[error] Error return value of `r.Body.Read` is not checked

(errcheck)